packages/dev/core/src/Meshes/Builders/greasedLineBuilder.ts

```diff
@@ -113,8 +114,16 @@ export interface GreasedLineMeshBuilderOptions extends GreasedLineMeshOptions {
 export function CreateGreasedLineMaterial(name: string, options: GreasedLineMaterialOptions, scene: Nullable<Scene>) {
     scene = <Scene>(scene ?? EngineStore.LastCreatedScene);
 
-    const material = options.materialType === GreasedLineMeshMaterialType.MATERIAL_TYPE_PBR ? new PBRMaterial(name, scene) : new StandardMaterial(name, scene);
-    new GreasedLinePluginMaterial(material, scene, options);
+    let material;
+    if (options.materialType === GreasedLineMeshMaterialType.MATERIAL_TYPE_PBR) {
+        material = new PBRMaterial(name, scene);
+        new GreasedLinePluginMaterial(material, scene, options);
+    } else if (options.materialType === GreasedLineMeshMaterialType.MATERIAL_TYPE_STANDARD) {
+        material = new StandardMaterial(name, scene);
+        new GreasedLinePluginMaterial(material, scene, options);
+    } else {
+        material = new GreasedLineSimpleMaterial(name, scene, options);
+    }

```

Comment:
two questions:

* (nit) this is a perfect case for `switch(options.materialType)`
* If we have no reference to the newly created `GreasedLinePluginMaterial`, how can it be disposed?

Reply:
Changed to a switch statement.

Reply:
It gets disposes automatically when the material which the plugin material is attached to is disposed.